### ISSUE TITLE
Read Postgres config values in dagster.yaml from env with schedule

### ISSUE THREAD

**Issue by ankxyz** [dagster-io/dagster#3079]
Hello.

I use **Postgres** as *event logs*, *run* and *schedule storage* (as here described https://docs.dagster.io/_apidocs/libraries/dagster_postgres).

Whey I execute pipeline(s) manually (in CLI or **Dagit** UI) everything is ok.

But if I start schedule its log shows me error: environment variables are not defined.

----

Example (my code and configuration):

`pipeline.py`
```python
import datetime
from datetime import datetime, time
from dagster import solid, pipeline, schedule, repository, daily_schedule


@solid
def say_hello(_):
    print('hello')


@pipeline
def say_hello_pipe():
    say_hello()


@schedule(cron_schedule="* * * * *", pipeline_name="say_hello_pipe")
def my_schedule(_):
    pass


@repository
def repo():
    return [say_hello_pipe, my_schedule]
```

`dagster.yaml`
```yaml
scheduler:
  module: dagster_cron.cron_scheduler
  class: SystemCronScheduler


event_log_storage:
  module: dagster_postgres.event_log
  class: PostgresEventLogStorage
  config:
    postgres_db:
      username:
        env: DAGSTER_DATABASE_USER
      password:
        env: DAGSTER_DATABASE_PASSWORD
      hostname:
        env: DAGSTER_DATABASE_HOST
      db_name:
        env: DAGSTER_DATABASE
      port:
        env: DAGSTER_DATABASE_PORT


run_storage:
  module: dagster_postgres.run_storage
  class: PostgresRunStorage
  config:
    postgres_db:
      username:
        env: DAGSTER_DATABASE_USER
      password:
        env: DAGSTER_DATABASE_PASSWORD
      hostname:
        env: DAGSTER_DATABASE_HOST
      db_name:
        env: DAGSTER_DATABASE
      port:
        env: DAGSTER_DATABASE_PORT


schedule_storage:
  module: dagster_postgres.schedule_storage
  class: PostgresScheduleStorage
  config:
    postgres_db:
      username:
        env: DAGSTER_DATABASE_USER
      password:
        env: DAGSTER_DATABASE_PASSWORD
      hostname:
        env: DAGSTER_DATABASE_HOST
      db_name:
        env: DAGSTER_DATABASE
      port:
        env: DAGSTER_DATABASE_PORT
```

`start commands`
```bash
export DAGSTER_HOME=$PWD
export PYTHONPATH=$PIPELINES_FOLDER
export DAGSTER_DATABASE_USER=DAGSTER
export DAGSTER_DATABASE_PASSWORD=DAGSTER
export DAGSTER_DATABASE=DAGSTER
export DAGSTER_DATABASE_HOST=localhost
export DAGSTER_DATABASE_PORT=5435

# Run Postgres

docker run \
    -d \
    --name=dagster-postgres \
    --rm \
    -e POSTGRES_USER=$DAGSTER_DATABASE_USER \
    -e POSTGRES_PASSWORD=$DAGSTER_DATABASE_PASSWORD \
    -e POSTGRES_DB=$DAGSTER_DATABASE \
    -p $DAGSTER_DATABASE_PORT:5432 \
    -v $PWD/ps_data:/var/lib/postgresql/data \
        postgres:12.2


dagster schedule up -f pipeline.py -d $DAGSTER_HOME
dagster schedule start -f pipeline.py -d $DAGSTER_HOME my_schedule
```

`Schedule scirpt generated by Dagster`
```bash

            #!/bin/bash
            export DAGSTER_HOME=/home/user/schedule_with_postgres
            export LANG=en_US.UTF-8
            

            export RUN_DATE=$(date "+%Y%m%dT%H%M%S")

            /usr/local/bin/python3.7 -m dagster api launch_scheduled_execution --schedule_name my_schedule -f /home/userschedule_with_postgres/pipeline.py -a repo -d /home/user/schedule_with_postgres "/home/user/schedule_with_postgres/schedules/logs/7b72dc0eefbc48bf1da9b11a7f4a9e2869015f96/${RUN_DATE}_7b72dc0eefbc48bf1da9b11a7f4a9e2869015f96.result"
        
```
---

As result I have schedule logs with this content:

```log
{"__class__": "IPCStartMessage"}
{"__class__": "IPCErrorMessage", "message": null, "serializable_error_info": {"__class__": "SerializableErrorInfo", "cause": null, "cls_name": "DagsterInvalidConfigError", "message": "dagster.core.errors.DagsterInvalidConfigError: Errors whilst loading configuration for <dagster.config.field_utils.Selector object at 0x7f2375b45a58>.\n    Error 1: Post processing at path root:postgres_db:username of original value {'env': 'DAGSTER_DATABASE_USER'} failed:\n(PostProcessingError) - dagster.config.errors.PostProcessingError: You have attempted to fetch the environment variable \"DAGSTER_DATABASE_USER\" which is not set. [/* and so on...*/]
{"__class__": "IPCEndMessage"}
```

---

In short, log says: **Postgres** config variables cannot be read because of corresponding environment variables are not defined.

On the one hand it's clear: in schedule script the environment variables are actually not defined. But on the other hand, I do not quite understand how to add env vars to this script: script is genereted by **Dagster**, but I did not found any parameters which allow to add something in the script.

Thank you for your attention.

**Comment by helloworld** [comment]
Hey @ankxyz, each of the schedule definition decorators takes an `environment_vars` argument that stores the variables in the script that is executed: https://docs.dagster.io/_apidocs/schedules#dagster.schedule. 

In your case, you can do something like:

```
@schedule(
    cron_schedule="* * * * *",
    pipeline_name="say_hello_pipe",
    environment_vars={
        key: os.environ.get(key)
        for key in [
            "DAGSTER_HOME",
            "PYTHONPATH",
            "DAGSTER_DATABASE_USER",
            "DAGSTER_DATABASE_PASSWORD",
            "DAGSTER_DATABASE",
            "DAGSTER_DATABASE_HOST",
            "DAGSTER_DATABASE_PORT",
        ]
        if key in os.environ
    },
)
def my_schedule(_):
    pass
```

We'll be coming out with a new scheduler soon in our 0.10.0 release that doesn't depend on cron and doesn't have this limitation.

**Comment by ankxyz** [comment]
@helloworld 
Yeah, it works!
Thank you! You helped me so much!
